### DISCUSSION TITLE
nuclei validates invalid templates (or, nuclei does not load semi-valid templates)

### DISCUSSION THREAD

**Discussion by MegaManSec** [projectdiscovery/nuclei#3001] [Q&A] [answered]
### Current Behavior:
Loading a template with an unknown tag will cause the template to be excluded.
Loading the template with -validate reports no errors

### Expected Behavior:
Either the template will be loaded, or -validate will report an error.

### Steps To Reproduce:
1. Run the following command and note the output:
```
nuclei -t fuzzing/wordpress-themes-detect.yaml -u 'https://1.1.1.1/' -debug

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v2.8.1

                projectdiscovery.io

[WRN] Could not load template /home/jrogers/nuclei-templates/fuzzing/wordpress-themes-detect.yaml: the template was excluded
[INF] Using Nuclei Engine 2.8.1 (latest)
[INF] Using Nuclei Templates 9.3.1 (latest)
[INF] Targets loaded for scan: 1
[INF] No results found. Better luck next time!
[FTL] Could not run nuclei: no valid templates were found
```

In the above, the template is not loaded.

We then use the -validate flag to check what's wrong with the template:
```
$ nuclei -t fuzzing/wordpress-themes-detect.yaml -u 'https://1.1.1.1/' -debug -validate

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v2.8.1

                projectdiscovery.io

[INF] All templates validated successfully
```
but no problems are discovered.

If we remove the tag "fuzz" from the yaml file (e.g.):
```
$ diff wordpress-themes-detect.yaml.old wordpress-themes-detect.yaml.new 
7c7
<   tags: fuzz,wordpress
---
>   tags: wordpress
```

the template is loaded successfully.

This issue currently affects the following templates, which AFAICT, are impossible to load at the moment:

```
cves/2017/CVE-2017-17562.yaml
cves/2019/CVE-2019-17382.yaml
cves/2020/CVE-2020-16139.yaml
cves/2022/CVE-2022-2034.yaml
default-logins/oracle/peoplesoft-default-login.yaml
fuzzing/adminer-panel-fuzz.yaml
fuzzing/cache-poisoning-fuzz.yaml
fuzzing/header-command-injection.yaml
fuzzing/iis-shortname.yaml
fuzzing/linux-lfi-fuzzing.yaml
fuzzing/mdb-database-file.yaml
fuzzing/prestashop-module-fuzz.yaml
fuzzing/wordpress-plugins-detect.yaml
fuzzing/wordpress-themes-detect.yaml
fuzzing/wordpress-weak-credentials.yaml
fuzzing/xff-403-bypass.yaml
grafana/grafana-file-read.yaml
miscellaneous/ntlm-directories.yaml
misconfiguration/gitlab/gitlab-user-enum.yaml
misconfiguration/proxy/open-proxy-internal.yaml
misconfiguration/proxy/open-proxy-localhost.yaml
misconfiguration/proxy/open-proxy-portscan.yaml
vulnerabilities/grafana/grafana-file-read.yaml
vulnerabilities/wordpress/wordpress-wp-cron.yaml
vulnerabilities/wordpress/wp-xmlrpc-brute-force.yaml
```

### Anything else:
This issue was discovered while creating https://github.com/projectdiscovery/nuclei-templates/pull/6291.

**Comment by ehsandeep** [comment, marked as answer]
@MegaManSec see https://nuclei.projectdiscovery.io/nuclei/get-started/#template-exclusion, templates with `fuzz` and `dos` tags are excluded from the default run and can be included with `-itags fuzz,dos`

**Reply by MegaManSec** [reply]
Aha, makes sense.

However, I would then argue that grafana-file-read.yaml is not a proper candidate for "fuzz". While it does involve a very small "fuzzing" component, the issue is inherently a path traversal vulnerability, which just relies on knowing at least one plugin which is installed (_many are installed by default_): [CVE-2021-43798](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-43798).
What are your thoughts?

**Reply by ehsandeep** [reply]
agreed, we should remove the `fuzz` tag from `grafana-file-read.yaml` template.
